include/aws/s3/s3_client.h

```diff
@@ -24,20 +24,57 @@ struct aws_s3_request;
 struct aws_s3_meta_request;
 struct aws_s3_meta_request_result;
 
+/**
+ * A Meta Request represents a group of generated requests that are being done on behalf of the
+ * original request. For example, one large GetObject request can be transformed into a series
+ * of ranged GetObject requests that are executed in parallel to improve throughput.
+ *
+ * The aws_s3_meta_request_type is a hint of transformation to be applied.
+ */
 enum aws_s3_meta_request_type {
+
+    /**
+     * The Default meta request type sends any request to S3 as-is (with no transformation). For example,
+     * it can be used to pass a CreateBucket request.
+     */
     AWS_S3_META_REQUEST_TYPE_DEFAULT,
+
+    /**
+     * The GetObject request will be split into a series of ranged GetObject requests that are
+     * executed in parallel to improve throughput, when possible.
+     */
     AWS_S3_META_REQUEST_TYPE_GET_OBJECT,
+
+    /**
+     * The PutObject request will be split into MultiPart uploads that are executed in parallel
+     * to improve throughput, when possible.
+     */
     AWS_S3_META_REQUEST_TYPE_PUT_OBJECT,
 
     AWS_S3_META_REQUEST_TYPE_MAX,
 };
 
+/**
+ * Invoked to provide response headers received during execution of the meta request.
+ * If an error prevents the execution of the HTTP request (e.g. error codes like
```

Comment:
hmmm, just thought about "If we don't ever send a request, there will not be headers for sure." So, suggestions:
```c
/**
 * Invoked to provide response headers received during execution of the meta request, both for success and  
 * error HTTP status codes.
 *
 * Return AWS_OP_SUCCESS to continue processing the request.
 * Return AWS_OP_ERR to indicate failure and cancel the request.
 */
```

Reply:
Fixed as suggested.